src/modules/gamewisp/global-emotes.js

```diff
@@ -0,0 +1,65 @@
+const $ = require('jquery');
+const settings = require('../../settings');
+
+const AbstractEmotes = require('../emotes/abstract-emotes');
+const Emote = require('../emotes/emote');
+
+const provider = {
+    id: 'gw-global',
+    displayName: 'GameWisp Global Emotes',
+    badge: 'https://d32y8axfzs6sv8.cloudfront.net/static/gamewisp_transparent_18px.png'
+};
+
+class GWGlobalEmotes extends AbstractEmotes {
+    constructor() {
+        super();
+
+        settings.add({
+            id: 'gwEmotes',
+            name: 'GameWisp Emotes',
+            defaultValue: false,
+            description: 'Use GameWisp emotes'
+        });
+
+        if (settings.get('gwEmotes')) {
+            this.updateGlobalEmotes();
+        }
+
+        settings.on('changed.gwEmotes', () => {
+            if (settings.get('gwEmotes') && !this.emotes.size) {
```

Comment:
this seems like buggy behavior.. if you already have the emotes in the map, you would want to update them on setting change probably.. since they are a map it should overwrite the existing ones anyways